### ISSUE TITLE
RPCClient doesn't work with long username / password

### ISSUE THREAD

**Issue by devinrsmith** [ConsensusJ/consensusj#28]
If the username / password is too long, `Base64.encodeToString` will input a newline somewhere inside of `basicAuth`. This is invalid for HTTP headers. Here is the relevant stacktrace and code part.

```
Exception in thread "main" java.lang.IllegalArgumentException: Illegal character(s) in message header value: Basic X19jb29raWVfXzo5MjJlNDQ3MDJjYWZiYzYwMzA1ZWViMDgyMTMzYjM2MTFlYjAxNzlhYTlhMjE4
Yzk0ODU2MmEwMjk4MmU0YzA=
	at sun.net.www.protocol.http.HttpURLConnection.checkMessageHeader(HttpURLConnection.java:541)
	at sun.net.www.protocol.http.HttpURLConnection.isExternalMessageHeaderAllowed(HttpURLConnection.java:492)
	at sun.net.www.protocol.http.HttpURLConnection.setRequestProperty(HttpURLConnection.java:3057)
	at com.msgilligan.jsonrpc.RPCClient.openConnection(RPCClient.java:175)
```

```java
...
String basicAuth = "Basic " + Base64.encodeToString(auth.getBytes(), 0).trim();
connection.setRequestProperty("Authorization", basicAuth);
...
```

**Comment by msgilligan** [comment]
This Base64 implementation (file) was copied from the Android project -- didn't want an external dependency, but wanted a solid implementation. Do you know if the issue had been reported or fixed upstream?

**Comment by devinrsmith** [comment]
A lot of base64 implementations insert newlines for "pretty-printing". It's not an issue w/ the Base64 implementation IMO. Also, it looks like Base64 is an internal library? `com.msgilligan.jsonrpc.util.Base64`. There is also `java.util.Base64` which obviously doesn't have any external dependencies, but I'm not sure what its status with newlines is. Worst case, it should be easy to use `String.replace("\n", "")`, or something similar.

**Comment by msgilligan** [comment]
It's internal, but copied (with copyright notice intact) from Android.

We needed to:
1. Have something that works on Java and Android
2. Avoid namespace conflicts
3. Avoid any external dependencies

**Comment by msgilligan** [comment]
[java.util.Base64](https://developer.android.com/reference/java/util/Base64.html) is added in *Android API level 26* (aka Android 8.0, "Oreo")

So, unless you're developing for Android 8.0 or later and you want the same code to work on Java and Android you have to provide your own implementation (or a factory, etc)

**Comment by msgilligan** [comment]
Fixed by c93571035799d99aece2e39d89d1737698c6995c -- on `master` branch -- will be in next release.
